sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -95,6 +95,12 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(math.abs(t1 - ct.getTime) < 5000)
   }
 
+  test("datetime function localtimestamp") {
+    val ct = LocalTimestamp().eval(EmptyRow).asInstanceOf[Long]
+    val t1 = DateTimeUtils.localDateTimeToMicros(LocalDateTime.now())
+    assert(math.abs(t1 - ct) < 5000)
+  }
```

Comment:
Could you check different combinations of the session time zone (the SQL config `(SQLConf.SESSION_LOCAL_TIMEZONE`) and the default (JVM) time zone (`DateTimeTestUtils.withDefaultTimeZone`). The list to check `DateTimeTestUtils.outstandingTimezonesIds` should be enough.